components/ActionSheet/src/MDCActionSheetController.m

```diff
@@ -0,0 +1,354 @@
+  return action;
+}
+
+@end
+
+@interface MDCActionSheetController () <MDCBottomSheetPresentationControllerDelegate,
+    UITableViewDelegate, UITableViewDataSource>
+
+@end
+
+@implementation MDCActionSheetController {
+  MDCActionSheetHeaderView *_header;
+  UITableView *_tableView;
+  MDCBottomSheetTransitionController *_transitionController;
+  NSMutableArray<MDCActionSheetAction *> *_actions;
+}
+
+@synthesize mdc_adjustsFontForContentSizeCategory = _mdc_adjustsFontForContentSizeCategory;
+
++ (instancetype)actionSheetControllerWithTitle:(NSString *)title message:(NSString *)message {
+  return [[MDCActionSheetController alloc] initWithTitle:title message:message];
+}
+
++ (instancetype)actionSheetControllerWithTitle:(NSString *)title {
+  return [MDCActionSheetController actionSheetControllerWithTitle:title message:nil];
+}
+
+- (instancetype)init {
+  return [MDCActionSheetController actionSheetControllerWithTitle:nil message:nil];
+}
+
+- (instancetype)initWithTitle:(NSString *)title message:(NSString *)message {
+  self = [super initWithNibName:nil bundle:nil];
+  if (self) {
+    _actions = [[NSMutableArray alloc] init];
+    _transitionController = [[MDCBottomSheetTransitionController alloc] init];
+    _transitionController.dismissOnBackgroundTap = YES;
+    /**
+     "We must call super because we've made the setters on this class unavailable and overridden
+     their implementations to throw assertions."
+     */
+    super.transitioningDelegate = _transitionController;
+    super.modalPresentationStyle = UIModalPresentationCustom;
+    _tableView = [[UITableView alloc] initWithFrame:CGRectZero style:UITableViewStylePlain];
+    _transitionController.trackingScrollView = _tableView;
+    _tableView.autoresizingMask = (UIViewAutoresizingFlexibleWidth
+                                   | UIViewAutoresizingFlexibleHeight);
+    _tableView.delegate = self;
+    _tableView.dataSource = self;
+    _tableView.estimatedRowHeight = 56.f;
+    _tableView.rowHeight = UITableViewAutomaticDimension;
+    _tableView.separatorStyle = UITableViewCellSeparatorStyleNone;
+    [_tableView registerClass:[MDCActionSheetItemTableViewCell class]
+       forCellReuseIdentifier:ReuseIdentifier];
+
+    _header = [[MDCActionSheetHeaderView alloc] initWithFrame:CGRectZero];
+    _header.title = [title copy];
+    _header.message = [message copy];
+    self.backgroundColor = [UIColor whiteColor];
+  }
+
+  return self;
+}
+
+- (void)addAction:(MDCActionSheetAction *)action {
+  [_actions addObject:action];
+  [self updateTable];
+}
+
+- (NSArray<MDCActionSheetAction *> *)actions {
+  return [_actions copy];
+}
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+
+  _tableView.frame = self.view.bounds;
+  [self.view addSubview:_tableView];
+  [self.view addSubview:_header];
+}
+
+- (void)viewWillLayoutSubviews {
+  [super viewWillLayoutSubviews];
+
+  CGSize size = [_header sizeThatFits:self.view.bounds.size];
```

Comment:
We can't directly access data members of a CGRect unless we know they have been standardized.  Note that this is a Obj-c only thing that is fixed in Swift.
If you want to access origin or size directly, you must first call:
CGRect standardizedBounds = CGRectStandardize(self.view.bounds);
If you are looking for height, or x or the like, and *don't* want to standardize, use the CGRectGetHeight() or CGRectGetMinX() functions.